src/org/freedesktop/gstreamer/GstObject.java

```diff
@@ -62,6 +70,76 @@ protected GstObject(Handle handle, boolean needRef) {
         this.handle = handle;
     }
 
+    /**
+     * Set the value of a GstObject property from a String representation.
+     * <p>
+     * The data value is deserialized using <code>gst_value_deserialize</code>.
+     *
+     * @param property the property to set
+     * @param data the value as a valid String representation
+     * @throws IllegalArgumentException if the data cannot be deserialized to
+     * the required type.
+     */
+    public void setAsString(String property, String data) {
+        GObjectAPI.GParamSpec propertySpec = findProperty(property);
+        if (propertySpec == null) {
+            throw new IllegalArgumentException("Unknown property: " + property);
+        }
+        final GType propType = propertySpec.value_type;
+
+        GValue propValue = new GValue();
+        GVALUE_API.g_value_init(propValue, propType);
+
+        boolean success = GSTVALUE_API.gst_value_deserialize(propValue, data);
+
+        if (success) {
+            GOBJECT_API.g_param_value_validate(propertySpec, propValue);
+            GOBJECT_API.g_object_set_property(this, property, propValue);
+        }
+
+        GVALUE_API.g_value_unset(propValue); // Release any memory
```

Comment:
Shouldn't a Cleanup like this better be placed in a finally-Block?
```java
try {
        GVALUE_API.g_value_init(propValue, propType);
        if(!GSTVALUE_API.gst_value_deserialize(propValue, data)) {
            throw new IllegalArgumentException(
                    "Unable to deserialize data to required type: "
                            + propType.getTypeName());
        }

        GOBJECT_API.g_param_value_validate(propertySpec, propValue);
        GOBJECT_API.g_object_set_property(this, property, propValue);
}
finally {
        GVALUE_API.g_value_unset(propValue); // Release any memory
}

Reply:
Probably not. If it fails anywhere here it's not necessarily recoverable. This is based on code elsewhere, and generally the bindings don't guard against errors in the native layers like this.

Reply:
Ok, your Code, your decision ;)

Reply:
Well, _final_ decision, and not for want of trying to get other maintainers! :wink: Thanks for the review - more eyes always appreciated. And if this wasn't calling into native layers I'd probably agree with you on this.